### PR TITLE
Move GAE environment setup to nox

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: system_tests/nox.py [hunk 1]
```diff
@@ -208,4 +214,37 @@ def session_compute_engine(session):
 
 def session_app_engine(session):
     session.virtualenv = False
-    session.run('pytest', 'app_engine/test_app_engine.py')
+
+    if SKIP_GAE_TEST_ENV in os.environ:
```
Comment (dhermes) [inline system_tests/nox.py]: But what if they set

```
export SKIP_APP_ENGINE_SYSTEM_TEST="Haha JK please run the GAE tests"
```

File: system_tests/nox.py [hunk 2]
```diff
@@ -208,4 +214,37 @@ def session_compute_engine(session):
...
+        session.log('Skipping App Engine tests.')
+        return
+
+    # Unlike the default tests above, the App Engine system test require a
+    # 'real' gcloud sdk installation that is configured to deploy to an
+    # app engine project.
+    # Grab the project ID from the cloud sdk.
+    project_id = subprocess.check_output([
+        'gcloud', 'config', 'list', 'project', '--format',
+        'value(core.project)']).strip()
```
Comment (dhermes) [inline system_tests/nox.py]: What if the command fails?

```python
>>> subprocess.check_output(['false'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/subprocess.py", line 574, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['false']' returned non-zero exit status 1
>>>
>>>
>>> subprocess.check_output(['not-installed'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/subprocess.py", line 567, in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

File: system_tests/test_app_engine.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+TEST_APP_URL = os.environ['TEST_APP_URL']
```
Comment (dhermes) [inline system_tests/test_app_engine.py]: Hi `KeyError`, I'm Dan

File: system_tests/nox.py (hunk 1 above)
Comment (jonparrott) [inline system_tests/nox.py]: Intentional, it's a common idiom that the presence of an environment variable is treated as 'truthy'.

File: system_tests/nox.py (hunk 2 above)
Comment (jonparrott) [inline system_tests/nox.py]: An error will be raised and nox will mark the session as failed. If you want, I can catch it and use `session.error`.

Comment (dhermes) [inline system_tests/nox.py]: `nox` marking session is fine

File: system_tests/test_app_engine.py (hunk 3 above)
Comment (dhermes) [inline system_tests/test_app_engine.py]: OK

**Comment by dhermes** [comment]
LGTM

**Pull request merged**